### DISCUSSION TITLE
Copilot Chat Now Generally Available on GitHub  🎉

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#147305] [Copilot News and Announcements]
We’re excited to announce that **[Copilot Chat](https://docs.github.com/en/copilot/using-github-copilot/asking-github-copilot-questions-in-github)** on GitHub is now generally available!  

These features are accessible to all users with the following plans:  
- **Copilot Free**: Includes access to Copilot Chat on GitHub.com.  
- **Copilot Pro** and **Copilot Business**: Include full access to Copilot Chat.  

## Manage Your Copilot Settings  
You can manage these features using the following resources:  
- [Copilot Settings for Free and Pro Users](https://docs.github.com/en/copilot/managing-copilot/managing-copilot-as-an-individual-subscriber/managing-copilot-policies-as-an-individual-subscriber#about-github-copilot-settings-on-github)  
- [Copilot Organization Policies for Business Users](https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/managing-policies-for-copilot-in-your-organization?tool=vscode#about-policies-for-github-copilot)  

## Updates to Terms  
These features are now covered under the standard GitHub Terms:  
- [GitHub Copilot Pro Terms](https://docs.github.com/en/site-policy/github-terms/github-terms-for-additional-products-and-features#github-copilot)  
- [GitHub Copilot Business Terms](https://github.com/customer-terms/github-copilot-product-specific-terms)  

**Note:** They are no longer governed by the GitHub Preview Terms.  

## Enhanced Copilot Chat Indexing  
To help you get the most out of Copilot Chat, we’ve removed the limits on the number of repositories you can **[index to improve Copilot Chat responses](https://docs.github.com/en/copilot/customizing-copilot/indexing-repositories-for-copilot-chat)**.  

## Thank You for Your Feedback  
Your insights during the [public preview](https://github.com/orgs/community/discussions/139872) were instrumental in improving these features. If you have additional feedback, please share it in the comments below. 

We can’t wait to see how these features enhance your development workflows! 🚀

**Comment by Eschguy** [comment]
How do I disable it?

**Reply by dangotbanned** [reply]
If you use uBlock, you can add these filters (https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters)

```
github.com##.copilotPreview__container
github.com##.AppHeader-CopilotChat
```

**Reply by craiglpeters** [reply]
For a pretty complete overview, take a look at [this answer](https://github.com/orgs/community/discussions/147437#discussioncomment-11628636).

**Comment by y-ack** [comment]
how do i disable it?

**Reply by y-ack** [reply]
[i now see a] settings page for whatever free advertier this is, and the dashboard prompt can be disabled: https://github.com/settings/copilot
![image](https://github.com/user-attachments/assets/930a1335-21a0-4d1f-8d30-863f49cde908)

(notice, however, that "Copilot in github.com" is still forced "Enabled", for cases of issues beyond the landing page intrusion)

**Reply by ghalfacree** [reply]
@y-ack  I don't have those options showing if I click that link - possibly because I have, very deliberately, never used Copilot.

This is all I have:

![Screenshot 2024-12-19 at 15-34-06 Copilot](https://github.com/user-attachments/assets/0a9254e3-e360-4b80-8e02-1c5797799b05)

Even though that would appear to suggest Copilot isn't enabled, it very much is: I still have the stupid chat box on the homepage.

**Reply by y-ack** [reply]
i specifically did not opt in to anything either, though i did send a nonsense question to the prompt eventually, so maybe that activated it??  i.e.
 - forced prompt on homepage, no setting to disable in default state
 - when you do interact with it, automatically activates 'copilot free' which enables the settings menu

very classy, github

**Reply by craiglpeters** [reply]
For a pretty thorough overview, see this [answer](https://github.com/orgs/community/discussions/147437#discussioncomment-11628636).

**Comment by michaelachrisco** [comment]
It looks like it can see into **private repos** within **private organizations** if your personal account has access. 

Not a huge deal if your account has access, but if an org wants certain repos to be non-trainable (for ACA/HIPAA/PII etc...) purposes, this could be an issue. Granted you should not be putting that stuff in git to begin with, but its better to opt out rather than opt in for legal purposes in my opinion. 

![image](https://github.com/user-attachments/assets/666f6bb9-f6dd-4a29-ac87-b6fdaadcf8d9)

**Reply by michaelachrisco** [reply]
Looks like you can disable it here:? https://docs.github.com/en/copilot/managing-copilot/managing-copilot-as-an-individual-subscriber/managing-your-github-copilot-pro-subscription/canceling-your-copilot-pro-trial-as-an-individual-user Im not sure

But it should still not be opt-in in my opinion.

**Reply by Foxboron** [reply]
None of those options tells me how to disable the Copilot prompt at the main dashboards.

**Reply by michaelachrisco** [reply]
> None of those options tells me how to disable the Copilot prompt at the main dashboards.

Nope I have no idea. The message is just there now.

**Reply by michaelachrisco** [reply]
![image](https://github.com/user-attachments/assets/3b22aed0-1be2-48ed-b258-ec6d83dfe8ee)

**Reply by michaelachrisco** [reply]
> None of those options tells me how to disable the Copilot prompt at the main dashboards.

![image](https://github.com/user-attachments/assets/6e870f99-c650-4e2e-b1bb-cd7f1fe80ca8)
Looks like you can disable the dashboard specific feature [here](https://github.com/settings/copilot).

**Reply by Foxboron** [reply]
I can't.

![image](https://github.com/user-attachments/assets/e879249d-89a2-4784-a414-4a6f9b9cfa2f)

**Reply by serebit** [reply]
I also don't see that option.

**Reply by michaelachrisco** [reply]
Interesting. I wonder if thats just because one of my orgs auto-added us? I have no idea why you dont have access.

**Reply by Akash1134** [reply]
Please note that Enterprises have the [option](https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/configuring-your-proxy-server-or-firewall-for-copilot) to block access to Free accounts, the same way you can block access to Copilot Individual (now Copilot Pro) users. If your company employs security measures like a firewall or proxy server, you should add the following URLs, ports, and protocols to an allowlist to ensure Copilot works as expected:

 * [api.githubcopilot.com](http://api.githubcopilot.com/)
 * [api.individual.githubcopilot.com](http://api.individual.githubcopilot.com/) 
 * [copilot-proxy.githubusercontent.com](http://copilot-proxy.githubusercontent.com/) 

Once you have these blocked in the firewall, users will not be able to use Copilot Free or Copilot Pro. 

Depending on the security policies and editors your organization uses, you may need to allowlist additional domains and URLs. For more information on specific editors, see [Further reading](https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/configuring-your-proxy-server-or-firewall-for-copilot#further-reading).

Every user of the proxy server or firewall also needs to configure their own environment to connect to Copilot. See [Configuring network settings for GitHub Copilot](https://docs.github.com/en/copilot/configuring-github-copilot/configuring-network-settings-for-github-copilot).

Hope that helps!

**Comment by Foxboron** [comment]
I can't stand this Copilot marketing.

It's currently taking up a whole 3 and a half centimeters in the dashboard (I measured this), if I click on anything there it redirects me to the marketing page to Github Copilot Pro!

This is effectively just a fucking ad for an AI product I'm woefully disinterested in. I can't even figure out how to disable this thing!

Make this thing opt-in and stop making the dashboard view worse then it already is.

**Reply by Foxboron** [reply]
If anyone reading this uses qutebrowser and desperately need to hide this element:

Append a user stylesheet to you config
```
c.content.user_stylesheets = ['~/.config/qutebrowser/user.css'] 
```

Create `user.css` with the following content.

```
.copilotPreview__container { display: none !important }
```

**Reply by Foxboron** [reply]
I'm apparently suppose to be able to disable it from [github.com/settings/copilot](https://github.com/settings/copilot), however I only have a "Try Copilot Free" ad there.

**Reply by ruthtillman** [reply]
Yeah, there would be a way to disable it if you were paying for it, but there seems to be no way to disable it (yet) from being an option at all.

Imagine launching this in December 2024 and still not realizing that tons of people would immediately want to remove it.

**Reply by craiglpeters** [reply]
For a pretty complete response, see [this answer](https://github.com/orgs/community/discussions/147437#discussioncomment-11628636).

**Reply by Foxboron** [reply]
@craiglpeters these options are not available to me.

**Comment by bytefish** [comment]
@Akash1134 Could you please share how to disable it? If there’s no way to disable it, this would also be a useful information.

**Reply by craiglpeters** [reply]
For a pretty complete response, see [this answer](https://github.com/orgs/community/discussions/147437#discussioncomment-11628636).

**Comment by michaelachrisco** [comment]
If a user is removed from an organization, will they still see any questions that they asked about a specific repo they previously had access to?Lets say I have a super secret repo that had some proprietary code within. Will the chat history still have that code? How long will the chat history have the code?

**Comment by serebit** [comment]
Somehow in your public preview nobody ever gave you the feedback that people might not want to see this on their homepage? Nobody suggested an option to disable it? Really?

Or, the more likely case, did management tell you to ignore those who don't want to see this forced AI slop on the homepage of a Git hosting site?

**Reply by joakim** [reply]
Must be some cargo cult in Microsoft where not wanting AI is blasphemy.

**Reply by tootbrute** [reply]
if they actually asked us if we wanted AI, it wouldn't exist.

**Comment by fossdd** [comment]
Please disable it and make it opt-in! This is horrible and just wants me to use a free GitHub alternative like https://codeberg.org instead without any AI ads whatsoever!

**Comment by ruthtillman** [comment]
Please supply documentation on how to disable it? I have seen things about how to disable if I were actually subscribing myself but not how to be sure I don't see it in either the site or VSCode (updated to add that I looked at VSCode and it looks like I'd have to sign up there, which is a start). Specifically, I don't want it popping up places, just in my settings as a thing I could turn on if I wanted.

**Comment by Dekkonot** [comment]
Let me disable this or I will leave GitHub. This isn't a negotiation, this is a statement of intent if you continue making your product measurably worse.

**Comment by lukaslueg** [comment]
There needs to be a way to disable the constant Copilot spam.

**Comment by higby** [comment]
Thank you for sending me the email notification regarding this change. I can only say that I'm disappointed by this as well as the homepage redesign shoving copilot further down my throat. I'm now actively transferring all my projects over to Codeberg.

**Comment by BigGunnLP** [comment]
AI sucks. This sucks. Stop it.

**Comment by lukaslueg** [comment]
One can delete the Copilot spam from homepage by blocking the `##.copilotPreview__container` and `##.AppHeader-CopilotChat` elements.
![1](https://github.com/user-attachments/assets/2afa601e-e6e6-4353-bb28-3d1f7a121e42)
![2](https://github.com/user-attachments/assets/0563f754-96db-4544-a44d-2d5d8ad5c080)

**Comment by tootbrute** [comment]
nobody wanted this.

> Microsoft: "Hey, we made silly AI Thing. Interested?"
Me: "No thanks"
Microsoft: "Try GitHub Copilot now! 1 Month free trial!"
Me: "Still no"
Microsoft": "HERE, JUST TAKE IT OKAY?"
*sends spam through all my e-mails, post, fax, pigeon post, ..*
>
> [#Microsoft](https://nattomaki.social/tags/Microsoft) [#GitHub](https://nattomaki.social/tags/GitHub) [#Copilot](https://nattomaki.social/tags/Copilot)
>
> By the way: [@Codeberg](https://social.anoxinon.de/@Codeberg) exists!
> from mastodon: https://nattomaki.social/notes/a1ykytrjdxainfyl

Also forgejo exists too. easy to setup guys.
https://forgejo.org/

**Comment by sinand99** [comment]
lol it is unbelievable that there are dinosaur developers here who refuse AI. Wake up dinosaurs! we are in the AI age. Use it or stay behind.

**Reply by chrede88** [reply]
By all means use it if it works for you. I think what most people is a bit angry about is the fact that Github pushes this on people without asking.

**Reply by tootbrute** [reply]
if it's so great, they wouldn't need to shove it down our throat.

**Comment by alvvaro** [comment]
How can it be disabled?

**Comment by SonoSooS** [comment]
uBlock Origin filter (optimized for speed), disables as many things as I could find.

```
||github.com/github-copilot/*$important
||github.githubassets.com/assets/*copilot-chat*.js
github.com##main div.news>:not(feed-container):style(display:none !important)
github.com##header div.AppHeader-CopilotChat:style(display:none !important)
```

**Comment by raster** [comment]
This is a great reminder that I should make sure all of my stuff is moved to CodeBerg because I don't want Copilot or any other AI that GitHub is offering.

**Comment by ghalfacree** [comment]
Adding my request to the above: please provide a way to fully disable Copilot.

**Comment by photex** [comment]
Please make a single and obvious way to disable Copilot. Better yet, remove Copilot entirely.

**Comment by prymitive** [comment]
I would also like to know how to get rid of the “Ask copilot” input that’s wasting my phones screen space without any benefit to me.
All the suggestions of prompts I see below it are things like “my open pull requests”, “recent commits in x/z” and other queries that I could have always easily get an answer to in a few clicks, without meting any glaciers or inflating global GPU prices, so it’s very hard to get excited about it.

**Comment by alvvaro** [comment]
![Screenshot 2024-12-20 at 08 17 09](https://github.com/user-attachments/assets/9f2d5036-c535-4f92-a942-2a9a2717d207)

The value proposition is ridiculous. You can find your issues quickly, or find your issues slowly while melting 100 times as many glaciers.

**Comment by jesus2099** [comment]
@Akash1134

I don't need this huge button.
I keep clicking it by mistake and it shows me useless things. 

It hides all the important info on mobile (user and repo names).

**Please remove copilot from my account.** 

The option is [forced enabled in the settings](https://github.com/settings/copilot) and I cannot disable it myself.

**Comment by Akash1134** [comment]
Hi everyone, 

Thanks for sharing your valuable feedback. We wanted to share more information on some of the concerns you have posted.  

### Removing Copilot from the main homepage
When you go to the main github.com dashboard, you see a Copilot chat bar at the top.  If you no longer want to see that select disable “Dashboard entry point”  in your [Copilot settings](https://github.com/settings/copilot) 
<img width="452" alt="Copilot Dashboard" src="https://github.com/user-attachments/assets/2ca82701-9092-4bad-8005-ae41e77504ce" />

Note that if you have your GitHub Copilot access provided to you via your company as a GitHub Copilot Business or Enterprise subscription then you will not see that setting as no prompt telemetry is collected and the policies around your usage are controlled by the organization administrator.

### Blocking GitHub Copilot in your network
Enterprise network administrators have the ability to block GitHub Copilot traffic at the network level. The endpoints used by Copilot are different depending on the subscription level.  Therefore it is possible to allow access for users with Business or Enterprise level (paid for by your organization), while blocking access for people using Pro or Free.  For more information see Managing GitHub [Copilot access to your organization's network](https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/managing-access-to-github-copilot-in-your-organization/managing-github-copilot-access-to-your-organizations-network)

### Disabling GitHub Copilot Telemetry Collection
If you are using Copilot Free or Copilot Pro, prompts from your editor and Copilot’s suggestions are shared with GitHub to help improve suggestions, related models and product features. [For more information see our privacy statement](https://docs.github.com/en/site-policy/privacy-policies/github-general-privacy-statement). If you want to continue using Copilot but disable this telemetry collection, uncheck the box “Allow GitHub to use my code snippets from the code editor for product improvements” in your [Copilot settings](https://github.com/settings/copilot).
<img width="452" alt="Telemetry Settings" src="https://github.com/user-attachments/assets/67c51294-b921-4486-8732-5a1548f145b1" />

### Switching off access to GitHub Copilot 
Finally - on GitHub, you will see the Copilot icon in the header as you browse around the site.  Copilot is only invoked when you use, i.e. if you use the chat window to ask Copilot questions.

Reading the community feedback, it sounds like some folks would prefer to never see that button.  Therefore we have decided to add the ability to turn the Copilot functionality off as a free user on GitHub. When that is rolled out you will be able to go into your Copilot settings and disable Copilot if you do not want to use it.  If you change your mind in the future then you will need to go back into your Copilot settings to enable it. Note that paid Copilot Pro users can already cancel their subscriptions from the settings control.

We currently don't have an ETA when that setting will be deployed yet but the team are actively working on it. We will report back when it is available.  In the meantime as long as you are not using Copilot via the chat window, it will not be active during your site visit.

Hope that helps, and please keep the feedback coming. ❤️

**Reply by serebit** [reply]
Thank you for responding to the feedback, Akash. My concern was that we would be ignored, and I'm glad that wasn't the case.

However, I still don't see the option to disable the prompt in my Copilot settings:

![image](https://github.com/user-attachments/assets/4dda5aca-9d61-481e-ada6-d19f3287e993)

uBlock Origin works for now, but I would like to know why this setting seems to only appear for some and not others.

**Reply by Akash1134** [reply]
Hi @serebit Are you referring to disable “Dashboard entry point” in your [Copilot settings](https://github.com/settings/copilot)? If so, please note that if you have your GitHub Copilot access provided to you via your company as a GitHub Copilot Business or Enterprise subscription then you will not see that setting as no prompt telemetry is collected and the policies around your usage are controlled by the organization administrator.

**Reply by Foxboron** [reply]
@Akash1134 the issue is that we can't click on the disabled button without pressing the "Start using Copilot Free" button.

**Reply by Akash1134** [reply]
@Foxboron The latest we have to share is being discussed in this [discussion](https://github.com/orgs/community/discussions/147437#discussioncomment-11628636).

**Reply by alvvaro** [reply]
@Akash1134 thank you for your response. I can understand (but not agree with) the corporate reason for forcing these features to everyone, but I cannot wrap my head around the fact that you did not even consider the possibility of people wanting to opt-out of it.

**Reply by serebit** [reply]
> Hi @serebit Are you referring to disable “Dashboard entry point” in your [Copilot settings](https://github.com/settings/copilot)? If so, please note that if you have your GitHub Copilot access provided to you via your company as a GitHub Copilot Business or Enterprise subscription then you will not see that setting as no prompt telemetry is collected and the policies around your usage are controlled by the organization administrator.

I don't believe that I have access to Copilot through a business. I have never used Copilot and I intend to never use Copilot. I have not clicked on any promotional Copilot material. Even if this were the case and I were under an administrator that gives me access, the setting to hide the prompt on the homepage *should still be available*. I believe most would agree.

**Comment by jesus2099** [comment]
Here is a userstyle solution for [Stylus](https://add0n.com/stylus.html), while waiting for the genuine OFF option:

https://github.com/xt0rted/github-nopilot

**Comment by uyriq** [comment]
How to access a chat room on github.com at https://github.com/copilot not from browser but from my own app? 
I'm wondering what api methods are available to get history and also to send a message to the chat and get a response.
Perhaps you can name react/native projects that implement an example of using this api?

**Reply by tallior** [reply]
I'm using the Copilot chat on GitHub.com.
The popup window for adding files as context is very small and doesn't expand to show the full path:

<img width="532" alt="image" src="https://github.com/user-attachments/assets/0e2e3b6a-9db8-4da9-95f2-b40bc5aa1bb6" />


When paths are short, it will display the entire path (including filename and extension), but when the file path inside the repo is too long, it will be cut, making it impossible to figure out which file it is.

**Comment by manuelmopur** [comment]
### Performance Observation

When enabling the "Dashboard Entry Point" feature, I noticed a significant decrease in performance while using GitHub, specifically when the chat functionality was activated. After disabling the feature, everything returned to normal and felt more responsive.

### System Specifications
- **Processor:** AMD Ryzen 7 3700X 8-Core Processor @ 3.59 GHz
- **RAM Installed:** 32.0 GB
- **Graphics:** AMD Radeon RX 5500 XT (Micro-Star International Co., Ltd. (MSI))
  - Technology: 7 nm
  - Memory: 8 GB GDDR6 (Micron)
  - ROPs / TMUs: 32 / 88

### Steps to Reproduce
1. Enable the "Dashboard Entry Point" feature.
2. Navigate GitHub and use the chat feature.
3. Observe performance degradation.

### Expected Behavior
GitHub should remain responsive even with the "Dashboard Entry Point" feature enabled.

### Actual Behavior
Performance slows down noticeably when the feature is active.

Please investigate the potential impact of this feature on user experience.

**Comment by raxod502** [comment]
![image](https://github.com/user-attachments/assets/e2f98ab0-386b-4c30-931a-d7a994fa094a)

Feedback: look at the screenshot above. It has a prominent button with the text `My open pull requests`. Since I navigated to the GitHub home page today in an attempt to get to my list of open pull requests (something I don't look at often), I immediately clicked that button.

For some reason that beggars belief, clicking this button takes you to an advertisement for an AI product that I have absolutely no interest in, and, moreover **has nothing whatsoever to do with the button I clicked**. Why in `$DEITY`'s name would I ever, *ever*, **ever** want to purchase an AI slop machine to **list my pull requests**? There's a goshdamned button for that already, it's in the top right. This is so disrespectful I don't understand how it even got through a single internal design review without being laughed out of the room.

I maintain many dozens of open-source projects, and it will be a drag to migrate all of them and all of my dependent communities and infrastructure over to a competitor. Frankly, I wasn't looking forward to the process, even though I suspected I would need to do it at some point. It's rather aggravating to me that you're actively pushing me to execute on that migration faster, and to start contacting the developers I support through GitHub Sponsors (around 20, at about $350/month) to request that they set up accounts on alternative donation platforms.

@Akash1134, please communicate to whoever is in charge of this project that it's actively, directly pushing current developers and maintainers off the GitHub platform. Please feel free to review https://github.com/radian-software to see some examples of the projects that will be going elsewhere if this doesn't turn around.

(By the way, is it *really* necessary that 100% of the "trending repositories" and 50% of the official announcements in that screenshot are for AI crud that I would never in a million years engage with in a way other than being induced to vomit? That just adds insult to injury.)

**Reply by kwshi** [reply]
i agree

**Comment by sayanarijit** [comment]
Feedback: It doesn't include wiki data when querying in a particular repo.

**Comment by alexandrecro** [comment]
Support .pdf and .docx files in Github Copilot Spaces
